### DISCUSSION TITLE
Writing a new Drawing tool and editor.

### DISCUSSION THREAD

**Discussion by treyspiva** [annotorious/annotorious#479] [General]
I was reading through the code for [annotorious-plugin-tools](https://github.com/annotorious/annotorious-plugin-tools). The tool and editor use Svelte. Is this a requirement? Are you able to use TypeScript to write a new tool and editor? 

While the `AnnotoriousOpenSeadragonAnnotator` and `AnnotoriousImageAnnotator` classes have the `registerDrawingTool` and `registerShapeEditor` members they are not documented. Is this because they are not supported?

**Comment by rsimon** [comment]
Yes, tools must be written in Svelte. (I'm sure there's a way to build them without it. But that would probably involve __a lot__ of troubleshooting, getting them to work with the rest of the system which, internally, is Svelte, too.)

`registerDrawingTool` and `registerShapeEditor` are not documented, simply, because I didn't have the time to do it.
